### PR TITLE
Add starknetjs compatibilty tests to CI

### PR DESCRIPTION
**Pull request by petscheit**
This introduces some tests to ensure compatibility with the starknet-js package. This PR completes #1148 

The corresponding testing code has been moved to[ this repo](https://github.com/petscheit/madara-js-tests)

# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:
- Testing

## What is the new behavior?
This PR introduces the following tests
- Declare and deploy v0 & v1 contracts to madara
- Read state from smart contacts & via provider api
- Invoke transactions & decode events
- Estimate fees for declare, deploy and invoke

### CODE REVIEW

**Comment by d-roak** [comment]
Let's not add the js in the Madara main repo, we can create an action in a separate repo and call it here in the workflow

**Comment by petscheit** [comment]
Sure thing, let me know to which repo I should add this

**Comment by petscheit** [comment]
Ok, I moved the tests to a different [repo](https://github.com/petscheit/madara-js-tests). 

One of the tests is failing, I don't really understand why that is the case. The fee is severely underestimated for contracts that are declared in genesis. This error does not pop up for contracts I declare during runtime and I check the fee estimation the same way across the different tests. How do we proceed here?

**Comment by d-roak** [comment]
cc @tdelabro 
also, can we create a repo for the action inside the keep-starknet-strange org?

File: .github/workflows/starknet-js-tests.yml
Comment (tdelabro) [inline .github/workflows/starknet-js-tests.yml:33]: It's more than just madara, it should be every sequencer.
`sequencer-starknet-js-tests`

File: .github/workflows/starknet-js-tests.yml
Comment (tdelabro) [inline .github/workflows/starknet-js-tests.yml:36]: Can you expose it as a github action we can call directly.
In the same way `cache` is exposed: https://github.com/actions/cache

**Review by tdelabro — CHANGES_REQUESTED** [review]

**Comment by tdelabro** [comment]
> can we create a repo for the action inside the keep-starknet-strange org?
yes we will when it's ready

File: .github/workflows/starknet-js-tests.yml
Comment (d-roak) [inline .github/workflows/starknet-js-tests.yml:30]: if u run `--dev` u don't need to specify the 
- `--tmp`
- `--rpc-external`
- `--rpc-methods`
- `--rpc-cors`

why are u limiting the pool and the number of peers? and why are u specifying `--no-telemetry`?

Comment (d-roak) [reply .github/workflows/starknet-js-tests.yml:30]:
```suggestion
          ./target/release/madara --dev --execution native &
```

**Review by d-roak — CHANGES_REQUESTED** [review]

File: .github/workflows/starknet-js-tests.yml
Comment (petscheit) [reply .github/workflows/starknet-js-tests.yml:30]: The blocks weren't sealing, so I eventually copied the contents from the run script. Also didn't want to burn through the workflow hours excessively

File: .github/workflows/starknet-js-tests.yml
Comment (petscheit) [reply .github/workflows/starknet-js-tests.yml:14]: copied from rpc tests, as im a workflow noob

File: .github/workflows/starknet-js-tests.yml
Comment (petscheit) [reply .github/workflows/starknet-js-tests.yml:29]: we can rename when I transfer the repo

File: .github/workflows/starknet-js-tests.yml
Comment (petscheit) [reply .github/workflows/starknet-js-tests.yml:32]: I gave this a shot, but no success. Could we create a separate ticket for this? Would like to get this PR merged and won't be able to finish the action this week.

**Comment by tdelabro** [comment]
@petscheit I would happily merge it in its current state, but the test is failing

**Review by tdelabro — APPROVED** [review]

**Comment by petscheit** [comment]
Was being super dumb on Friday, I just pushed the action, so lets include this one also. 

Regarding the failing test, yup. You need to tell me how to proceed, as I'm measuring the results, and the fee estimation seems to be wrong for genesis contracts.

**Comment by tdelabro** [comment]
@d-roak can you review and approve ?

**Comment by tdelabro** [comment]
@petscheit https://github.com/keep-starknet-strange/madara/actions/runs/6611182918
The tests are not runing anymore. Github action config files are broken

**Comment by petscheit** [comment]
Hmm so the issue is that my action is a workflow correct? And doesn't seem to work. This is what I was struggling with on Friday. Any ideas how to get around this?

File: .github/workflows/pull-request.yml
Comment (d-roak) [inline .github/workflows/pull-request.yml:44]: if u are directly using the `petscheit/madara-js-tests@main`, why do we need the `starknet-js-tests.yml`?

Comment (d-roak) [reply .github/workflows/pull-request.yml:44]: just saw that u are not even allowed to do that

Comment (d-roak) [inline .github/workflows/pull-request.yml:44]:
```suggestion
    uses: ./.github/workflows/starknet-js-tests.yml
```

File: .github/workflows/push.yml
Comment (d-roak) [inline .github/workflows/push.yml:35]:
```suggestion
    uses: ./.github/workflows/starknet-js-tests.yml
```

**Review by d-roak — COMMENTED** [review]
applying both suggestions should fix it

**Comment by d-roak** [comment]
we won't merge it until it's green like a lettuce 🥬 
@tdelabro any clue why we are having issues with the estimated fees?

**Comment by tdelabro** [comment]
nope. @petscheit  started investigating it. But he will be busy IRL soon, so he can open and issue with his current advancement and someone will pick it up from there

**Comment by petscheit** [comment]
OK issue fixed, nothing wrong with madara. 

starknetjs didnt document that you have to pass the constructor calldata for the deploy estimation ( I thought it would be derived via ABI). When I add it, everything works correctly. I also opened a PR on starknetjs to update the docs.

**Comment by tdelabro** [comment]
@d-roak do you approve?

**Review by d-roak — APPROVED** [review]
On this side it's good. The changelog line should be added either in the beginning or the end, but since it's not standardized, the end result will be the same

I would feel more comfortable using the `keep-starknet-strange/madara-js-tests` before merging, but if u don't think it's necessary, u can merge it @tdelabro

**Comment by petscheit** [comment]
yeah we can also transfer the repo now, and update the repo link in this PR.

**Comment by tdelabro** [comment]
@petscheit ok lets do this. Ping me on tg when you did the transfer request

**Pull request merged**
